launcher/ui/MainWindow.cpp

```diff
@@ -1290,6 +1301,40 @@ void MainWindow::updateToolsMenu()
     ui->actionLaunchInstance->setMenu(launchMenu);
 }
 
+void MainWindow::updateThemeMenu()
+{
+    QMenu *themeMenu = ui->actionChangeTheme->menu();
+
+    if (themeMenu) {
+        themeMenu->clear();
+    } else {
+        themeMenu = new QMenu(this);
+    }
+
+    auto themes = APPLICATION->getValidApplicationThemes();
+
+    QActionGroup* ThemesGroup = new QActionGroup( this );
+
+    for (int i = 0; i < themes.size(); i++)
+    {
+        auto *theme = themes[i];
+        QAction * themeAction = themeMenu->addAction(theme->name());
```

Comment:
We usually add new actions to `all_actions` so that they get retranslated if the user changes their language. Not sure if a QMenu will automatically retranslate all their sub-actions.

Reply:
Well, it retranslate as the user change the language via settings (the only way to do this while the launcher is opened I think?). I also update everything when the settings window is closed, check [a12f892](https://github.com/PrismLauncher/PrismLauncher/pull/454/commits/a12f892841399163fbc12e72c61c7ed6fae882cf)

Anyway, works on my machine :trollface: